### PR TITLE
HPCC-22563 Replace legacy audience with appropriate audience

### PR DESCRIPTION
**Pull request by shamser**
Signed-off-by: Shamser Ahmed <shamser.ahmed@lexisnexis.co.uk>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [x] Test my draft Pull Request.

### CODE REVIEW

**Comment by shamser** [comment]
@ghalliday Please can you review

**Comment by ghalliday** [comment]
There are a few (5) calls in the code in the LN repository that also need replacing.

File: plugins/fileservices/fileservices.cpp [hunk 1]
```diff
@@ -732,7 +732,7 @@ static void blockUntilComplete(const char * label, IClientFileSpray &server, ICo
 
 FILESERVICES_API char * FILESERVICES_CALL implementSprayFixed(ICodeContext *ctx, const char * sourceIP, const char * sourcePath, int recordSize, const char * destinationGroup, const char * destinationLogicalName, int timeOut, const char * espServerIpPort, int maxConnections, bool overwrite, bool replicate, bool compress, bool failIfNoSourceFile, int expireDays, const char * dfuServerQueue)
 {
-    PrintLog("Spray:  %s", destinationLogicalName);
+	LOG(MCauditInfo, "Spray:  %s", destinationLogicalName);
```
Comment (ghalliday) [inline plugins/fileservices/fileservices.cpp]: tab used for indentation, rather than spaces

File: plugins/fileservices/fileservices.cpp [hunk 2]
```diff
@@ -820,7 +820,7 @@ FILESERVICES_API char * FILESERVICES_CALL fsfSprayFixed_v3(ICodeContext *ctx, co
 
 static char * implementSprayVariable(ICodeContext *ctx, const char * sourceIP, const char * sourcePath, int sourceMaxRecordSize, const char * sourceCsvSeparate, const char * sourceCsvTerminate, const char * sourceCsvQuote, const char * sourceCsvEscape, const char * destinationGroup, const char * destinationLogicalName, int timeOut, const char * espServerIpPort, int maxConnections, bool overwrite, bool replicate, bool compress, bool failIfNoSourceFile, bool recordStructurePresent, bool quotedTerminator, const char * encoding, int expireDays, const char * dfuServerQueue)
 {
-    PrintLog("Spray:  %s", destinationLogicalName);
+	LOG(MCauditInfo, "Spray:  %s", destinationLogicalName);
```
Comment (ghalliday) [inline plugins/fileservices/fileservices.cpp]: indent: tab->spaces

File: plugins/fileservices/fileservices.cpp [hunk 3]
```diff
@@ -958,7 +958,7 @@ FILESERVICES_API char * FILESERVICES_CALL fsfSprayVariable_v7(ICodeContext *ctx,
 
 FILESERVICES_API char * FILESERVICES_CALL implementSprayXml(ICodeContext *ctx, const char * sourceIP, const char * sourcePath, int sourceMaxRecordSize, const char *sourceRowTag, const char *sourceEncoding, const char * destinationGroup, const char * destinationLogicalName, int timeOut, const char * espServerIpPort, int maxConnections, bool overwrite, bool replicate, bool compress, bool failIfNoSourceFile, int expireDays, const char * dfuServerQueue)
 {
-    PrintLog("Spray:  %s", destinationLogicalName);
+	LOG(MCauditInfo, "Spray:  %s", destinationLogicalName);
 
     CClientFileSpray server;
     Owned<IConstWorkUnit> wu = getWorkunit(ctx);
```
Comment (ghalliday) [inline plugins/fileservices/fileservices.cpp]: indent: tab->spaces

File: plugins/fileservices/fileservices.cpp [hunk 4]
```diff
@@ -2074,7 +2074,7 @@ FILESERVICES_API char * FILESERVICES_CALL fsfRemotePull(ICodeContext *ctx,
                                                      bool wrap,
                                                      bool compress)
 {
-    PrintLog("RemotePull(%s):  %s%s", remoteEspFsURL,sourceLogicalName,asSuperfile?" as superfile":"");
+	LOG(MCauditInfo, "RemotePull(%s):  %s%s", remoteEspFsURL,sourceLogicalName,asSuperfile?" as superfile":"");
```
Comment (ghalliday) [inline plugins/fileservices/fileservices.cpp]: indent: Tab

**Comment by ghalliday** [comment]
@jakesmith happy with the audit logging?  It probably makes sense.
@shamser some indentation issues, and PR required for LN repo.

File: plugins/fileservices/fileservices.cpp [hunk 5]
```diff
@@ -1190,7 +1190,7 @@ FILESERVICES_API void FILESERVICES_CALL fsReplicate(ICodeContext *ctx, const cha
 
 FILESERVICES_API char * FILESERVICES_CALL fsfReplicate(ICodeContext *ctx, const char * sourceLogicalName, int timeOut, const char * espServerIpPort)
 {
-    PrintLog("REPLICATE:  %s", sourceLogicalName);
+    DBGLOG("REPLICATE:  %s", sourceLogicalName);
```
Comment (jakesmith) [inline plugins/fileservices/fileservices.cpp]: This one should be MCauditInfo too I think.

**Comment by jakesmith** [comment]
@shamser - added 1 comment.

**Comment by shamser** [comment]
@jakesmith @ghalliday Code modified as suggested.  Here's the LN PR: https://github.com/hpcc-systems/LN/pull/932

**Comment by jakesmith** [comment]
@shamser - There's another one fsDeleteLogicalFile (that currently uses DBGLOG / didn't change in this PR), but like Replicate, should now also be MCauditInfo.
But there are other (file service calls) that don't log at all, to be consistent they should audit too - could be done in another PR though.

**Comment by shamser** [comment]
@jakesmith I'm going through fileservices again and checking other cases where MCauditInfo should be used.

**Comment by shamser** [comment]
In line 426, there is a call to AuditMessage() which produces log messages in the format:
FileAccess,FileServices,DeleteSuperFile,W20190717-125645,regress,regress::single::roxie::w20190717-125645::myemptysuperfile"

@jakesmith Do we need an additional audit log line in fsDeleteLogicalFile()?

**Comment by jakesmith** [comment]
> Do we need an additional audit log line in fsDeleteLogicalFile()?

@shamser - not sure, AuditMessage ends up calling:
```
    LOG(daliAuditLogCat,"%s",aln.str());
```

is that equivalent to the new MCauditInfo use cases?

**Comment by shamser** [comment]
Their definitions  are identical.  Of course the daliAuditLogCat, implies that this one is dali related.  I wouldn't think the fileservices audit messages would be considered dali related but it does seem to be using **dali**AuditLogCat.  What do you think should we get rid of one of them (daliAuditLogCat or MCauditInfo) or is there a purpose to keep them separate?

**Comment by jakesmith** [comment]
The reason it's called 'daliAuditLogCat' is because all audit audience logging issued by components that have issued connectLogMsgManagerToDali(), will pipe that logging to Dali and it will be recorded in the Dali audit logging file.

So MCauditInfo will now also go there, so  the changed fileservice logging in this PR will go there too (worth verifying that it actually does).
I think that's a good thing (especially as some already did via using daliAuditLogCat).

I think we should consolidate on MCauditInfo (get rid of daliAuditLogCat).

**Comment by jakesmith** [comment]
@shamser - back to you.

**Comment by shamser** [comment]
@jakesmith I replaced daliAuditLogCat with MCauditInfo and tested to make sure it's being logged to mydali/audit/DaAudit.....log

File: dali/datest/datest.cpp [hunk 6]
```diff
@@ -3309,7 +3309,7 @@ void testMultiConnect()
 }
 
 #if 0
-LogMsgCategory const daliAuditLogCat(MSGAUD_audit, MSGCLS_information, DefaultDetail);
+LogMsgCategory const MCauditInfo(MSGAUD_audit, MSGCLS_information, DefaultDetail);
```
Comment (jakesmith) [inline dali/datest/datest.cpp]: @shamser - this shouldn't be needed, it's already declared / will be same as one in jlog.

**Comment by jakesmith** [comment]
@shamser - 1 comment

**Comment by shamser** [comment]
@jakesmith I've fixed that.  Shall I squash?

**Comment by jakesmith** [comment]
>Shall I squash?

@shamser - yes, please.

**Comment by shamser** [comment]
@jakesmith

**Comment by jakesmith** [comment]
@shamser - looks good.

@ghalliday @richardkchapman - ready to merge.

**Pull request merged**